inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java

```diff
@@ -38,29 +46,118 @@ public SortClientConfig getConfig() {
+    }
+
+    public void addCallBackFail(InLongTopic topic, int partitionId, int count, long time) {
+        SortSdkMetricItem metricItem = this.getMetricItem(topic, partitionId);
+        metricItem.callbackFailCount.addAndGet(count);
+        metricItem.callbackTimeCost.addAndGet(time);
+    }
+
+    public void addAckSuccess(InLongTopic topic, int partitionId) {
+        SortSdkMetricItem metricItem = this.getMetricItem(topic, partitionId);
+        metricItem.ackSuccCount.incrementAndGet();
+    }
+
+    public void addAckFail(InLongTopic topic, int partitionId) {
+        SortSdkMetricItem metricItem = this.getMetricItem(topic, partitionId);
+        metricItem.ackFailCount.incrementAndGet();
     }
 
-    public SortClientStateCounter getStateCounterByTopic(InLongTopic topic) {
-        return statManager.getStatistics(config.getSortTaskId(),
-                topic.getInLongCluster().getClusterId(), topic.getTopic());
+    public void addTopicOnlineCount(int count) {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.topicOnlineCount.addAndGet(count);
     }
 
-    public SortClientStateCounter getDefaultStateCounter() {
-        return statManager.getStatistics(config.getSortTaskId());
+    public void addTopicOfflineCount(int count) {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.topicOfflineCount.addAndGet(count);
+    }
+
+    public void addRequestManager() {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerCount.incrementAndGet();
+    }
+
+    public void addRequestManagerFail(long time) {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerFailCount.incrementAndGet();
+        metricItem.requestManagerTimeCost.addAndGet(time);
+    }
+
+    public void addRequestManagerConfChange() {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerConfChangedCount.incrementAndGet();
+    }
+
+    public void addRequestManagerCommonError() {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerCommonErrorCount.incrementAndGet();
+    }
+
+    public void addRequestManagerParamError() {
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerParamErrorCount.incrementAndGet();
+    }
+
+    private SortSdkMetricItem getMetricItem(InLongTopic topic, int partitionId) {
+        Map<String, String> dimensions = new HashMap<>();
+        dimensions.put(SortSdkMetricItem.KEY_SORT_TASK_ID, sortTaskId);
+        if (topic != null) {
+            dimensions.put(SortSdkMetricItem.KEY_CLUSTER_ID, topic.getInLongCluster().getClusterId());
+            dimensions.put(SortSdkMetricItem.KEY_TOPIC_ID, topic.getTopic());
+        }
+        dimensions.put(SortSdkMetricItem.KEY_PARTITION_ID, String.valueOf(partitionId));
```

Comment:
Add a configuration item for partitionId dimension.

Reply:
fixed, thx